aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -856,7 +856,7 @@ std::tuple<Tensor, Tensor, Tensor> _lu_with_info_cuda(const Tensor& self, bool p
   squareCheckInputs(self);
   auto req_size = self.sizes().vec();
   req_size.pop_back();
-  Tensor pivots_tensor = at::zeros(req_size, self.options().dtype(at::kInt));
+  Tensor pivots_tensor = at::arange(1, self.size(-1) + 1, self.options().dtype(at::kInt)).expand(req_size);
```

Comment:
I think we need to expand and clone rather than just expand. `lu_solve` requires `pivots` to be contiguous, and expanding this tensor to higher dimensions makes it non-contiguous. Additionally, this is also used in the MAGMA routines, so having expanded tensors will cause memory to overwritten.

Reply:
Hmm, would it be simpler if you just put these changes directly into my fork for? I just added you as a collaborator there so you should be able to push to this branch

Reply:
Great, thanks!